src/fairseq2/utils/read_and_pickle_wrap_iterator.py

```diff
@@ -0,0 +1,48 @@
+from typing import Callable, Iterator, Tuple, TypeVar
```

Comment:
We recently moved to Python 3.10 and updated the code base to use some of the modern syntax features that came with 3.9 and 3.10.

We now import `Callable` and `Iterator` from `collections.abc`, and instead of `Tuple`, `List`, and `Dict`, we now use `tuple`, `list`, and `dict` in typing annotations (e.g. `foo: tuple[int, str]`). You can check out the code base to see their uses.

It would be nice if you could make the relevant changes in this PR as well. I see a few `Tuple`s at least that can be changed to `tuple`.